src/core/model.jl

```diff
@@ -190,13 +194,56 @@ Return an iterator over all agents of the model.
 """
 allagents(model) = values(model.agents)
 
-
 """
     allids(model)
 Return an iterator over all agent IDs of the model.
 """
 allids(model) = keys(model.agents)
 
+#######################################################################################
+# %% Higher order collections
+#######################################################################################
+export iter_agent_groups, map_agent_groups, index_groups_filtered
+
+"""
+    iter_agent_groups(order::Int, model::ABM)
+
+Return an iterator over all agents of the model, grouped by order. When `order = 2`, the
+iterator returns agent pairs, e.g `(agent1, agent2)` and when `order = 3`: agent triples,
+e.g. `(agent1, agent7, agent8)`. `order` must be larger than `1` but has no upper bound.
+
+Index order is provided by the [`by_id`](@ref) scheduler.
```

Comment:
I mean, here it should be trivial to allow any scheduler instead of forcing the `by_id`, right? Not that it matters much